src/rules/require-repository-directory.js

```diff
@@ -0,0 +1,17 @@
+const LintIssue = require('./../LintIssue');
+
+const lintId = 'require-repository-directory';
+const nodeName = 'repository';
+const message = 'repository object missing directory property';
+const ruleType = 'standard';
+
+const lint = (packageJsonData, severity) => {
+  if (!packageJsonData[nodeName].hasOwnProperty('directory')) {
```

Comment:
Thank you, @greatislander!! Would you mind adding a check that `repository` exists? We could hit a scenario where the rule is enabled, but the property doesn't exist in the package.json file.

Reply:
@tclindner Makes sense, will do!

Reply:
Thank you!!